pommerman/network/ion_client/network.py

```diff
@@ -0,0 +1,186 @@
+                raise Exception(constants.Exceptions.net_server_full.value)
+            raise Exception(constants.Exceptions.net_respond_fail.value)
+        except:
+            self.lock.release()
+            raise Exception(constants.Exceptions.net_connect_fail.value)
+
+    def join_list(self, room=False):
+        """Argument:  
+        * room: The room to be created/joined. If False, the public room will \
+be joined
+        Description: Check if the server actually responds"""
+        self.lock.acquire()
+        if not room:
+            self.ws_.send(
+                rapidjson.dumps(
+                    {
+                        "intent": constants.NetworkCommands.match.value
+                    }
+                )
+            )
+        else:
+            self.ws_.send(
+                rapidjson.dumps(
+                    {
+                        "intent": constants.NetworkCommands.room.value,
+                        "room": room
+                    }
+                )
+            )
+        self.lock.release()
+        decoded_msg = rapidjson.loads(self.ws_.recv())
+        if (
+            decoded_msg["intent"] == constants.NetworkCommands.status_full.value
+        ):
+            if room:
+                raise Exception(constants.Exceptions.room_full.value)
+            else:
+                raise Exception(constants.Exceptions.match_full.value)
+        self.id = decoded_msg["player_id"]
+        self.mode = decoded_msg["mode"]
+
+    def waitm(self):
+        """Description: Wait for a response from the server regarding a match 
+        request"""
+        self.lock.acquire()
+        try:
+            r = rapidjson.loads(self.ws_.recv())
+            if r["intent"] == constants.NetworkCommands.match_start.value:
+                self.match_id = r["match_id"]
+        except:
+            self.lock.release()
+            raise Exception(constants.Exceptions.net_server_closed.value)
+        self.lock.release()
+
+    def match_get(self):
+        """Description: Get the next step of the match"""
+        self.lock.acquire()
+        try:
+            message_recieved = self.ws_.recv()
+        except:
+            self.lock.release()
+            raise Exception(constants.Exceptions.net_respond_fail.value)
+        self.lock.release()
+        try:
+            r = rapidjson.loads(str(gzip.decompress(message_recieved), "utf-8"))
+        except:
+            try:
+                message_recieved = rapidjson.loads(message_recieved)
+                if message_recieved[
+                    "intent"
+                ] == constants.NetworkCommands.match_end.value:
+                    return [
+                        2,
+                        int(message_recieved["reward"]),
+                        int(message_recieved["agent"])
+                    ]
+            except:
+                raise Exception(constants.Exceptions.net_invalid_response.value)
+        if r["d"]:
+            return [1]
```

Comment:
Clarify.

Reply:
Done

Reply:
What does `return [1]` mean? This is unnecessarily obfuscated.

Reply:
(Clarify meant everything about this)

Reply:
I'll fix this with an Enum.

Reply:
The integers denote different responses. I've added some context in the docstring here's a copy.
```Return values(0th element of array):
* 0 - Agent is still alive - OBS and Turn ID
* 1 - Agent is dead - Nothing (Agent is dead)
* 2 - Match has ended - Reward and agent item ID correlating to pommerman.constants.Items```

Reply:
This is exactly what an enum is meant to address.

Reply:
I don't like micro-enums all that much.